devito/ops/compiler.py

```diff
@@ -30,3 +53,130 @@ def prepare_ops(self, soname, ccode, hcode):
         else:
             warning("Couldn't find OPS_INSTALL_PATH \
                 environment variable, please check your OPS installation")
+
+    def cuda_compiler(self, soname):
+        # CUDA kernel compilation
+        cuda_src = '%s/CUDA/%s_kernels.cu' % (self.get_jit_dir(), soname)
+        cuda_target = '%s/%s_kernels_cu' % (self.get_jit_dir(), soname)
+
+        cuda_code = ""
+        try:
+            with open(cuda_src, 'r') as f:
+                cuda_code = f.read()
+        except FileNotFoundError:
+            raise ValueError("The file %s isn't present" % cuda_src)
+
+        cuda_device_compiler = CUDADeviceCompiler()
+        cuda_host_compiler = CudaHostCompiler()
+
+        with warnings.catch_warnings():
+            warnings.simplefilter('ignore')
+
+            # Spinlock in case of MPI
+            sleep_delay = 0 if configuration['mpi'] else 1
+            _, _, cuda_o, _ = compile_from_string(
+                cuda_device_compiler, cuda_target,
+                cuda_code, cuda_src,
+                cache_dir=self.cache_dir,
+                debug=True,
+                sleep_delay=sleep_delay,
+                object=True
+            )
+            _, _, src_o, _ = compile_from_string(
+                cuda_host_compiler, self.target,
+                self.code, self.ops_src,
+                cache_dir=self.cache_dir,
+                debug=True,
+                sleep_delay=sleep_delay,
+                object=True
+            )
+            cuda_host_compiler.link_extension(
+                '%s%s' % (self.target, cuda_host_compiler.so_ext),
+                [src_o, cuda_o],
+                debug=True
+            )
+
+    def jit_compile(self, soname):
+        from devito import clear_cache
+        clear_cache()
+
+        self.target = str(self.get_jit_dir().joinpath(soname))
+        self.ops_src = '%s/%s_ops.cpp' % (self.get_jit_dir(), soname)
+        self.cache_dir = self.get_codepy_dir().joinpath(soname[:7])
```

Comment:
can you remember me why `[:7]` here

Reply:
Just a temporary directory. Its name is created using the beginning of the hash.
Equals to devito/compile.py.